### ISSUE TITLE
Dragging should maintain the cursor position relative to the bar

### ISSUE THREAD

**Issue by davidje13** [nathancahill/Split.js#112]
Currently when dragging, the gutter snaps so that its centre lines up with the cursor. When working with large gutters this is not desirable, as it causes an unexpected jump. This is especially problematic when users want to make fine adjustments to the arrangement.

It should be possible (at least via an option and I would suggest by default) to drag gutters based on where the cursor clicks on them. This would be easy to implement by adjusting `start` within `startDragging` (immediately after calling `calculateSizes` https://github.com/nathancahill/Split.js/blob/master/split.js#L366). The change required would be:

```
start += mouseLocation - gutterCentre;
```

**Comment by nathancahill** [comment]
Sounds great. Can you submit a PR?

**Comment by davidje13** [comment]
Sure, I'll have a go at some point this weekend. What are your general contribution guidelines here? (testing / browser testing / linting / minifying / CI / etc.?)

**Comment by nathancahill** [comment]
Testing: The initialization options are [tested with Jasmine](https://github.com/nathancahill/Split.js/blob/master/test/split.spec.js). The interaction is not tested yet, but I'd love to implement it some day with [Leaflet/prosthetic-hand](https://github.com/Leaflet/prosthetic-hand).

Browser testing. Test in whatever browsers you have access to. Ideally Firefox and Chrome, Safari if Mac, Edge if Windows.

Linting: There's an eslint config in the root of the project.

CI: BrowserStack sponsers CI testing across all supported browsers. https://github.com/nathancahill/Split.js#browser-support

Minifying: This happens when a new build is published, so only submit your PR with changes in `split.js`.

Keep in mind that the goals of the repo (besides being the best split view utility) are tiny code size and zero dependencies.

Thanks!

**Comment by davidje13** [comment]
@nathancahill OK, that all sounds good (btw I suggest you put all that in a CONTRIBUTING.md file for future contributors)

Can you confirm which version of eslint you're using? When I check out the code as-is I get a few linter errors (using eslint 3.19.0), but if I forcibly downgrade it I get errors due to unrecognised configuration options.

Alternatively I can fix most of the 3.19.0 issues easily enough, except that it's complaining about "Assignment to property of function parameter 'el'" in `setElementSize` (and similar for `setGutterSize`), but clearly changing the elements is the purpose of those functions.

**Comment by nathancahill** [comment]
I'm away from my computer at the moment, but you can throw a eslint disable line comment for now.

Added you as contributor on the repo, you seem to have a good grasp as to what needs to be done.

**Comment by davidje13** [comment]
I have implemented this with

```
        if (!dragMiddle) {
            const percentage = a.size + b.size
            const ratio = self.size / percentage
            self.start = getMousePosition(e) - (a.size * ratio)
        }
```

at the end of `startDragging`. `getMousePosition` is a function extracted from the existing `drag` function:

```
    function getMousePosition (e) {
        // Supports touch events, but not multitouch, so only the first
        // finger `touches[0]` is counted.
        if ('touches' in e) {
            return e.touches[0][clientAxis]
        }
        return e[clientAxis]
    }
```

This works fine except that it's revealed an existing issue where some handles will be slightly offset from their correct position (e.g. try dragging the right-most bar of the bottom example at https://nathancahill.github.io/Split.js/; notice how it snaps so the cursor is on the right of the bar rather than the centre). I'm looking through the code at the moment to see why this might be the case.

I'll push this to a branch in my fork & make a PR when I'm happy with it.

**Comment by nathancahill** [comment]
Addressed in #156.

**Comment by nathancahill** [comment]
Published in v1.5.4.
